### PR TITLE
Added login API support for AST

### PR DESCRIPTION
**Pull request by alex-ko-dev**
### Description

Previously, as a temporary solution, AST-SAST client used a ready-made access token. Such token had to be generated manually (e.g. in Postman).

AST-SAST client is now able to create access tokens itself. To do this, client ID and client secret need to be provided to the client. These fields were added to com.checkmarx.sdk.config.AstProperties and can be now specified in YAML config.

### References

User story [218](https://dev.azure.com/CxFlow/CxFlow/_workitems/edit/218/)

### Testing

Negative test (i.e. using wrong credentials) has been added to astScanProcessing.feature.
Positive test (login with valid credentials) is implicitly included into AST scan scenario.

### CODE REVIEW

**Review by olgakil — APPROVED** [review]

File: src/test/java/com/checkmarx/flow/cucumber/integration/ast/scans/AstRemoteRepoScanSteps.java [hunk 1]
```diff
@@ -171,49 +203,58 @@ public void setScanInitiator(String initiatorList) {
...
+        int actualSize = actualFindings.size();
+        String message = String.format("Expected that %s finding count would be %s, but got %d.",
+                scannerType,
+                expectedCountDescr,
+                actualSize);
+        boolean countIsExpected = expectedCountDescr.equals("0") ? actualSize == 0 : actualSize > 0;
+        assertTrue(message, countIsExpected);
+    }
 
     public void startScan(List<VulnerabilityScanner> scanners) {
```
Comment (ofersk) [inline src/test/java/com/checkmarx/flow/cucumber/integration/ast/scans/AstRemoteRepoScanSteps.java:238]: this function can use the isEnabled, it does not need to get any parameter

File: src/test/resources/cucumber/features/integrationTests/ast/astScanProcessing.feature [hunk 2]
```diff
@@ -1,35 +1,44 @@
...
 
-  @ASTRemoteRepoScan 
-  Scenario Outline: scan multiple initiators
-    Given scan initiator list is "<initiators>"
-    Then the returned contain populated results for all initiators
+
+  @ASTRemoteRepoScan
+  Scenario Outline: using multiple vulnerability scanners
+    Given enabled vulnerability scanners are "<scanners>"
+    Then scan results contain populated results for all scanners
```
Comment (ofersk) [inline src/test/resources/cucumber/features/integrationTests/ast/astScanProcessing.feature:8]: when running scan

Given is for setup
When is for performing an action
Then is for validating results

File: src/test/resources/cucumber/features/integrationTests/ast/astScanProcessing.feature [hunk 3]
```diff
@@ -1,35 +1,44 @@
...
     And sca finding count will be "<sca_findings>" and ast findings count "<ast_findings>" will be accordingly
```
Comment (ofersk) [inline src/test/resources/cucumber/features/integrationTests/ast/astScanProcessing.feature:9]: this is the then

File: src/test/resources/cucumber/features/integrationTests/ast/astScanProcessing.feature [hunk 4]
```diff
@@ -1,35 +1,44 @@
...
+      | AST      | 0            | > 0          |
+      | AST,SCA  | > 0          | > 0          |
 
-  @ASTRemoteRepoScan
-  Scenario Outline: check error message on expired AST token
-    Given AST scan with expired token
-    Then expired token expected error will be returned "<message>"
+  @ASTRemoteRepoScan @InvalidCredentials
+  Scenario Outline: Trying to scan with invalid credentials
+    When CxFlow tries to start AST scan with the "<client id>" and "<client secret>" credentials
```
Comment (ofersk) [inline src/test/resources/cucumber/features/integrationTests/ast/astScanProcessing.feature:20]: Given AST scan with the "<client id>" and "<client secret>" credentials
When running scan

File: src/test/resources/cucumber/features/integrationTests/ast/astScanProcessing.feature [hunk 5]
```diff
@@ -1,35 +1,44 @@
...
+      | <valid-client-id>  | <empty>         | AST client secret wasn't provided |
+      | <empty>            | <valid-secret>  | AST client ID wasn't provided     |
+
 
   @ASTRemoteRepoScan
-  Scenario Outline: check error message when AST is down
-    Given AST scan is initiated when AST is not available
-    Then unavailable AST server expected error will be returned "<message>"
+  Scenario Outline: AST is not accessible
+    When AST scan is initiated with API url: "<url>"
```
Comment (ofersk) [inline src/test/resources/cucumber/features/integrationTests/ast/astScanProcessing.feature:33]: this is a given

**Review by ofersk — COMMENTED** [review]
there are some places were you fixed bad code but not all the way.

please look at my comments starting with the comments on the feature.

**Pull request merged**